specfile/sources.py

```diff
@@ -100,14 +101,26 @@ def _extract_number(self) -> Optional[str]:
     @property
     def number(self) -> int:
         """Source number."""
-        return self._number or int(self._extract_number() or 0)
+        if self._number is not None:
+            return self._number
+        return int(self._extract_number() or 0)
 
     @property
     def number_digits(self) -> int:
-        """Number of digits in the source number."""
-        if self._number:
+        """
+        Gets number of digits in the source number.
+
+        Returns 0 if the source has no number, 1 if the source number
+        has no leading zeros and the actual number of digits if there are
+        any leading zeros.
+        """
+        if self._number is not None:
             return 0
-        return len(self._extract_number() or "")
+        number = self._extract_number()
+        if not number:
+            return 0
+        zeros = list(itertools.takewhile(lambda c: c == "0", number))
```

Comment:
Wouldn't `number.startswith("0")` suffice? We don't need the actual number of zeros.

Reply:
You are right, I'll fix that.